src/main/java/com/coreos/jetcd/resolver/URIResolvers.java

```diff
@@ -0,0 +1,116 @@
+package com.coreos.jetcd.resolver;
+
+import com.coreos.jetcd.exception.EtcdExceptionFactory;
+import com.google.auto.service.AutoService;
+import io.grpc.EquivalentAddressGroup;
+import java.net.InetSocketAddress;
+import java.net.URI;
+import java.util.Collections;
+import java.util.Hashtable;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import javax.naming.NamingEnumeration;
+import javax.naming.directory.Attributes;
+import javax.naming.directory.DirContext;
+import javax.naming.directory.InitialDirContext;
+
+public final class URIResolvers {
+
+  private URIResolvers() {
+  }
+
+  @AutoService(URIResolver.class)
+  public static final class Direct implements URIResolver {
+    private ConcurrentMap<URI, List<EquivalentAddressGroup>> cache;
+
+    public Direct() {
+      this.cache = new ConcurrentHashMap<>();
+    }
+
+    @Override
+    public int priority() {
+      return Integer.MIN_VALUE;
+    }
+
+    @Override
+    public boolean supports(URI uri) {
+      return "http".equals(uri.getScheme()) || "https".equals(uri.getScheme());
+    }
+
+    @Override
+    public List<EquivalentAddressGroup> resolve(URI uri) {
```

Comment:
should we also add a check on `uri` to see if it is supported in `resolve`? If not supported then don't add it the the group and cache. Right now it seems to me that the caller can by pass in whatever `uri` that it wants if it forgets to check the `uri` against `supports`.